lib/httpClient.js

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2022 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+'use strict'
+
+const axios = require('axios')
+const nodeUrl = require('url')
+const httpsClient = require('https')
+const Pino = require('pino')
+
+class HttpClient {
+  constructor(baseUrl, requestHeadersToProxy, baseOptions = {}, logger) {
+    this.baseUrl = baseUrl
+    this.requestHeadersToProxy = requestHeadersToProxy
+    this.baseOptions = baseOptions
+    this.logger = logger || Pino({ level: 'silent' })
+
+    const httpsAgent = getHttpsAgent(baseOptions)
+    this.axios = axios.create({
+      ...httpsAgent ? { httpsAgent } : {},
+      timeout: baseOptions.timeout,
+    })
+  }
+
+  async get(path, options = {}) {
+    return this.makeCall(path, undefined, {
+      ...options,
+      method: 'GET',
+    })
+  }
+
+  async post(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'POST',
+    })
+  }
+
+  async put(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'PUT',
+    })
+  }
+
+  async patch(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'PATCH',
+    })
+  }
+
+  async delete(path, payload, options = {}) {
+    return this.makeCall(path, payload, {
+      ...options,
+      method: 'DELETE',
+    })
+  }
+
+  async makeCall(path, payload, options) {
+    let headers = getHeaders(options, this.requestHeadersToProxy, this.baseOptions)
+    if (typeof payload === 'object') {
+      headers = {
+        'content-type': 'application/json;charset=utf-8',
```

Comment:
Can you just append the property instead of using the spread operator?

So you can declare it as const.

Reply:
Improved the handle of headers